### PR TITLE
channel: add backoff configuration

### PR DESCRIPTION
**Pull request by BusyJay**

### CODE REVIEW

File: src/channel.rs [hunk 1]
```diff
@@ -12,11 +12,12 @@
 // limitations under the License.
 
 
-use std::{mem, ptr};
+use std::{cmp, mem, ptr, usize};
```
Comment (disksing) [inline src/channel.rs]: Why usize need to be explicitly imported?

File: src/channel.rs [hunk 2]
```diff
@@ -50,6 +53,11 @@ fn format_user_agent_string(agent: &str) -> CString {
     CString::new(val).unwrap()
 }
 
+fn dur_to_ms(dur: Duration) -> usize {
+    let micro = dur.as_secs() * 1000 + dur.subsec_nanos() as u64 / 1_000_000;
```
Comment (disksing) [inline src/channel.rs]: millisecond?

File: src/channel.rs (hunk 1 above)
Comment (BusyJay) [inline src/channel.rs]: So I can use `usize::MAX`.

**Comment by disksing** [comment]
LGTM.

**Pull request merged**